static-instrumenter/agent-instrumenter/src/main/java/io/opentelemetry/contrib/staticinstrumenter/util/AgentCreator.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter.util;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+import io.opentelemetry.contrib.staticinstrumenter.util.path.AgentPathGetter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+import java.util.jar.JarOutputStream;
+import java.util.jar.Manifest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Creates a new agent JAR. */
+public final class AgentCreator {
+
+  private static final Logger logger = LoggerFactory.getLogger(AgentCreator.class);
+  private final TmpDirManager dirManager;
+  private final AgentPathGetter pathGetter;
+
+  public AgentCreator(TmpDirManager dirManager, AgentPathGetter pathGetter) {
+    this.dirManager = dirManager;
+    this.pathGetter = pathGetter;
+  }
+
+  public void create(Path finalJarPath, File otelJar, Map<String, byte[]> classesToInstrument)
+      throws IOException {
+
+    String tmpPath = dirManager.getTmpFile("agent-jars", "agent-", ".jar").toString();
+
+    Map<String, byte[]> classesToInstrumentCopy = new HashMap<>(classesToInstrument);
+
+    // following code mimics ByteBuddy's DynamicType.inject()
+    // we don't use it directly, since it would mean copying jar every time we add one class
+    try (JarInputStream in = new JarInputStream(new FileInputStream(otelJar))) {
+
+      Manifest manifest = in.getManifest();
+
+      try (JarOutputStream zout =
+          manifest == null
+              ? new JarOutputStream(new FileOutputStream(tmpPath))
+              : new JarOutputStream(new FileOutputStream(tmpPath), manifest)) {
```

Comment:
Instead of using `FileInputStream`/`FileOutputStream` and strings, you can use `Files.newInputStream()`/`Files.newOutputStream()` and `Path`s